diesel/src/pg/connection/mod.rs

```diff
@@ -1,113 +1,223 @@
-mod cursor;
-pub mod raw;
-#[doc(hidden)]
-pub mod result;
-mod row;
-mod stmt;
-
-use std::ffi::CString;
-use std::os::raw as libc;
-
-use self::cursor::*;
-use self::raw::RawConnection;
-use self::result::PgResult;
-use self::stmt::Statement;
-use crate::connection::*;
-use crate::deserialize::{Queryable, QueryableByName};
-use crate::pg::{metadata_lookup::PgMetadataCache, Pg, PgMetadataLookup, TransactionBuilder};
+use super::{Pg, PgMetadataLookup, PgValue};
+use crate::connection::{
+    AnsiTransactionManager, Connection, MaybeCached, SimpleConnection, StatementCache,
+};
+use crate::deserialize::{FromSqlRow, Queryable, QueryableByName};
+use crate::pg::metadata_lookup::PgMetadataCache;
+use crate::pg::transaction::TransactionBuilder;
 use crate::query_builder::bind_collector::RawBytesBindCollector;
-use crate::query_builder::*;
-use crate::result::ConnectionError::CouldntSetupConfiguration;
-use crate::result::*;
+use crate::query_builder::{AsQuery, QueryFragment, QueryId};
+use crate::result::ConnectionError::{self, CouldntSetupConfiguration};
+use crate::result::Error::DeserializationError;
+use crate::result::{ConnectionResult, QueryResult};
+use crate::row::{NamedRow, Row};
 use crate::sql_types::HasSqlType;
-
-/// The connection string expected by `PgConnection::establish`
-/// should be a PostgreSQL connection string, as documented at
-/// <https://www.postgresql.org/docs/9.4/static/libpq-connect.html#LIBPQ-CONNSTRING>
+use bytes::BytesMut;
+use postgresql::fallible_iterator::FallibleIterator;
+use postgresql::types::Type;
+use postgresql::Statement;
+use std::cell::RefCell;
+use std::convert::TryFrom;
+use std::error::Error;
+
+/// This connection type is based on the pure rust postgresql implementation
+/// inside the [postgres](https://docs.rs/postgres/0.17.0/postgres/index.html) crate.
+///
+/// See the documentation on [`postgres::Client`](https://docs.rs/postgres/0.17.0/postgres/struct.Client.html#method.connect)
+/// for connection strings that are accepted by `PgConnection::establish`
+/// or use the the `TryFrom` impl to create a new `PgConnection` from
+/// an existing `postgres::Client`
 #[allow(missing_debug_implementations)]
 pub struct PgConnection {
-    pub(crate) raw_connection: RawConnection,
+    conn: RefCell<postgresql::Client>,
```

Comment:
This creates a large_enum_variant which makes clippy unhappy.
Suggestions ?

Reply:
Just wrapping the inner connection into a box should make clippy happy here.